src/architectures.jl

```diff
@@ -2,7 +2,7 @@
 abstract type AbstractArchitecture end
 
 array_type(::KA.CPU) = Array
-array_type(::KA.CUDADevice) = CUDA.CuArray
+array_type(::KA.GPU) = CUDA.CuArray
```

Comment:
not sure we want to fallback on `CuArray` here?

Reply:
It is for now everywhere in the code. As I described above, this should work for CUDA and CPU only. I think we should wait until we move something actually to AMDGPU.jl.